### PR TITLE
Literal typing for indexer methods

### PR DESCRIPTION
**Pull request by emptymalei**
- [ ] closes https://github.com/noatamir/pydata-workshop/issues/11 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

😄 
@phofl @noatamir @jorisvandenbossche @marcogorelli

### CODE REVIEW

**Review by twoertwein — APPROVED** [review]
Assuming mypy&pyright are happy, this looks good to me.

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1921,7 +1921,7 @@ def sortlevel(
         level=None,
         ascending: bool | list[bool] = True,
         sort_remaining=None,
-        na_position: str_t = "first",
+        na_position: Literal["first", "last"] = "first",
```
Comment (twoertwein) [inline pandas/core/indexes/base.py:1924]: Could use `na_position: NaPosition = "first",`
https://github.com/pandas-dev/pandas/blob/c85bbc6913b9c4d2acc53c54cc858c84bf0658d1/pandas/_typing.py#L362

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -5573,7 +5573,7 @@ def sort_values(
         self,
         return_indexer: bool = False,
         ascending: bool = True,
-        na_position: str_t = "last",
+        na_position: Literal["first", "last"] = "last",
```
Comment (twoertwein) [inline pandas/core/indexes/base.py:5576]: Same as above

File: pandas/core/indexes/range.py [hunk 3]
```diff
@@ -493,7 +494,7 @@ def sort_values(
         self,
         return_indexer: bool = False,
         ascending: bool = True,
-        na_position: str = "last",
+        na_position: Literal["first", "last"] = "last",
```
Comment (twoertwein) [inline pandas/core/indexes/range.py:497]: and here again

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (emptymalei) [reply pandas/core/indexes/base.py]: Thanks for the review. This makes a lot of sense. Addressed in b1ef125

Comment (emptymalei) [reply pandas/core/indexes/base.py]: Hmm got a strange error in Minimum Version check.

```
2023-04-19T05:22:59.5815494Z error    libmamba Bad conversion of configurable 'remote_connect_timeout_secs' from source '~/.condarc' : yaml-cpp: error at line 14, column 30: bad conversion
```

This doesn't seem to be related to the change?

**Comment by twoertwein** [comment]
Thanks @emptymalei!

**Comment by emptymalei** [comment]
Thank you @twoertwein

**Pull request merged**